lms/views/admin/email.py

```diff
@@ -0,0 +1,44 @@
+from pyramid.httpexceptions import HTTPBadRequest, HTTPFound
+from pyramid.view import view_config, view_defaults
+
+from lms.security import Permissions
+from lms.tasks.email_digests import send_instructor_email_digests
+
+
+@view_defaults(route_name="admin.email", permission=Permissions.ADMIN)
+class AdminEmailViews:
+    def __init__(self, request):
+        self.request = request
+
+    @view_config(
+        route_name="admin.email",
+        request_method="GET",
+        renderer="lms:templates/admin/email.html.jinja2",
+    )
+    def get(self):
+        return {}
+
+    @view_config(route_name="admin.email", request_method="POST")
+    def post(self):
+        to_email = self.request.POST["to_email"]
+        h_userids = self.request.POST["h_userids"].split()
+        since = self.request.POST["since"]
+        until = self.request.POST["until"]
+
+        if not to_email.endswith("@hypothes.is"):
+            raise HTTPBadRequest(
+                "Test emails can only be sent to @hypothes.is addresses."
+            )
+
+        if len(h_userids) > 3:
+            raise HTTPBadRequest(
+                "Test emails can only be sent for up to 3 users at once."
+            )
```

Comment:
These just lead to ugly crash pages. In future we could replace this with reloading the admin page with a nice flash message error box.

Reply:
You can also use marshmallow schema to get the validation errors


https://github.com/hypothesis/lms/blob/add-email-digests-admin-page/lms/views/admin/application_instance/create.py#L14

And then use `from lms.views.admin import flash_validation`